src/library/baseplaylistfeature.cpp

```diff
@@ -402,18 +410,18 @@ void BasePlaylistFeature::slotCreateImportPlaylist() {
     }
 
     // Set last import directory
-    QFileInfo fileName(playlist_files.first());
-    m_pConfig->set(ConfigKey("[Library]", "LastImportExportPlaylistDirectory"),
-            ConfigValue(fileName.dir().absolutePath()));
+    QFileInfo fileInfo(playlist_files.first());
+    m_pConfig->set(kConfigKeyLastImportExportPlaylistDirectory,
+            ConfigValue(fileInfo.dir().absolutePath()));
```

Comment:
The explanation for this change got lost somehow:
The point of this change is that I don't really see a reason to reuse the `fileInfo` variable inside the loop. So inlining it here and instead declaring a const version in the loop seems like better style to me.

Reply:
`playlist_files.first()` is a QString, so this won't work.

Reply:
Sorry yes, noticed after the fact too. We can just construct a temporary QFileInfo (I updated the code suggestion accordingly). This pattern of getting the absolutePath of the folder containing some file has also been used multiple times. We might want to consider factoring that out into an utility function. Especially since I don't think this entire chain of `QString -> QFileInfo -> QDir -> QString` is particularly efficient.

Reply:
I'm aware this chain is the opposite of 'simple', though I think refactoring should happen in main where there are more places where this is needed (effect chain import/export comes to mind).
I'll add a ToDo.

I suppose to extract the dir string we just need to chop everything after the last [QDir::separator](https://doc.qt.io/qt-5/qdir.html#separator).

Reply:
```cpp
str.truncate(str.lastIndexOf(QDir::separator));
```

Reply:
if we are sure that the path's we get are absolute, yes that should be sufficient.